./configure fails on detecting libboost_program_options

According to

http://doc.powerdns.com/compiling-powerdns.html

" If your operating system does not have a Boost package, you don't need to compile all of boost just for PowerDNS. PowerDNS only uses Boost include files, so there is no need to install all of boost. Just untar the Boost distribution file and point instruct ./configure to find it, perhaps like this:

	$ CXXFLAGS=-I/home/bert/download/boost_1_33_0 ./configure ..."

but the ./configure step fails thus:

setenv CXXCPP /usr/sfw/bin/cpp ; ./configure --prefix=/opt/powerdns --with-boost="${HOME}/devel/src/boost_1_38_0" CC=gcc CXX=g++

checking for Boost headers version >= 103400... /export/home/user/devel/src/boost_1_38_0
checking for Boost's header version... 1_38
checking boost/foreach.hpp usability... yes
checking boost/foreach.hpp presence... yes
checking for boost/foreach.hpp... yes
checking for the toolset name used by Boost for g++... gcc34 -gcc
checking boost/program_options.hpp usability... yes
checking boost/program_options.hpp presence... yes
checking for boost/program_options.hpp... yes
checking for the Boost program_options library... no
configure: error: cannot not find the flags to link with Boost program_options

So the documentation states libraries are not needed, yet the ./configure looks for libboost_program_options-*gcc*.so.

There is a bit of code in ./configure with a series of checks which appears to be the root cause.

OS is SunOS 5.10 u3 (Solaris 10 11/06). The error occurs on both i86pc and sparc platforms.

What do I need to do to fix this?

pdns 3.4.5 fails to detect Boost properly:

checking for Boost headers version >= 1.35.0... yes
checking for Boost's header version... 
configure: error: invalid value: boost_major_version=''

Boost version is still 1.38.0 (/opt/abcd/include/boost-1_38).  ./configure is called with the following Boost switches:

--with-boost=/opt/abcd/include/boost-1_38 --enable-static-boost

CPPFLAGS="-I/opt/abcd/include/boost-1_38 -I/opt/abcd/include"; export CPPFLAGS

however, the above worked in 3.4.1, with those same settings above:
checking for Boost headers version >= 1.35.0... /opt/abcd/include/boost-1_38
checking for Boost's header version... 1_38
checking boost/foreach.hpp usability... yes
checking boost/foreach.hpp presence... yes
checking for boost/foreach.hpp... yes
checking for the toolset name used by Boost for g++... gcc47 -gcc
checking boost/program_options.hpp usability... yes
checking boost/program_options.hpp presence... yes
checking for boost/program_options.hpp... yes
checking for the Boost program_options library... yes
checking boost/archive/text_oarchive.hpp usability... yes
checking boost/archive/text_oarchive.hpp presence... yes
checking for boost/archive/text_oarchive.hpp... yes
checking for the Boost serialization library... (cached) yes

--- a/m4/boost.m4
+++ b/m4/boost.m4
@@ -90,7 +90,6 @@ dnl So get rid of the # lines, and glue the remaining ones together.
 (eval "$ac_cpp conftest.$ac_ext") 2>&AS_MESSAGE_LOG_FD |
   grep -v '#' |
   tr -d '\r' |
-  tr -s '\n' ' ' |
   $SED -n -e "$1" >conftest.i 2>&1],
   [$3],
   [$4])

